src/Cache/Cache.php

```diff
@@ -113,6 +114,26 @@ class Cache
     protected static $_registry;
 
     /**
+     * Returns the Cache Registry instance used for creating and using cache adapters.
+     * Also allows for injecting of a new registry instance.
+     *
+     * @param \Cake\Core\ObjectRegistry $registry Injectable registry object.
+     * @return \Cake\Cache\CacheRegistry
+     */
+    public static function registry(ObjectRegistry $registry = null)
```

Comment:
Yeah, there actually is a legitimate use case for a custom one. Although it could also extend CacheRegistry. That class does not implement any other methods, though so I think hinting with the abstract class is fine